src/app/icd/ICDManager.cpp

```diff
@@ -228,5 +254,74 @@ void ICDManager::OnTransitionToIdle(System::Layer * aLayer, void * appState)
     pIcdManager->mStateObserver->OnTransitionToIdle();
 }
 
+bool ICDManager::EnsureChipStackLock()
+{
+    // TODO implement platform agnostic lock creation/mechanism that can be used for ICD without locking the chip stack
+    bool requireLock = false;
+
+#if CHIP_STACK_LOCK_TRACKING_ENABLED
+    requireLock = !DeviceLayer::PlatformMgr().IsChipStackLockedByCurrentThread();
+#endif
+
+    if (requireLock)
+    {
+        DeviceLayer::PlatformMgr().LockChipStack();
+    }
+    return requireLock;
+}
+
+/* ICDNotify observer functions. */
+void ICDManager::KeepActiveRequest(KeepActiveFlags request, bool set)
+{
+    bool requiresUnlock = this->EnsureChipStackLock();
+
+    if (request == KeepActiveFlags::kExchangeContextOpen)
+    {
+        // There can be multiple open exchange context at the same time.
+        // Keep track of the requests count.
+        if (set)
+        {
+            this->OpenExchangeContextCount++;
+            this->SetKeepActiveModeRequirements(request, set);
```

Comment:
since we know set is true here, would it be more readable to place ` true /* set */` here  and maybe `false /* set */` below?